lib/ecto/model/timestamps.ex

```diff
@@ -23,11 +23,23 @@ defmodule Ecto.Model.Timestamps do
     if get_change changeset, field do
       changeset
     else
-      {date, {h, m, s}} = :erlang.universaltime
-      put_change changeset, field, Ecto.Type.load!(type, {date, {h, m, s, 0}})
+      put_change changeset, field, Ecto.Type.load!(type, timestamp_tuple_with_usec(changeset.repo.adapter))
     end
   end
 
+  defp timestamp_tuple_with_usec(Ecto.Adapters.MySQL) do
```

Comment:
To correctly implement this function, we must have to know what MySQL version is being used, right? Not sure how it would be implemented and if it's going to be efficient.

Another solution would be passing the `precision` option to these date types in the migration, instead of manually setting it to `datetime(6)`.